plugins/jabra-file/fu-jabra-file-device.c

```diff
@@ -36,13 +38,48 @@ fu_jabra_file_device_to_string(FuDevice *device, guint idt, GString *str)
 	fwupd_codec_string_append_hex(str, idt, "DfuPid", self->dfu_pid);
 }
 
+static gboolean
+fu_jabra_file_device_probe(FuDevice *device, GError **error)
+{
+	FuJabraFileDevice *self = FU_JABRA_FILE_DEVICE(device);
+	FuUsbInterface *iface = NULL;
+	FuUsbEndpoint *ep1 = NULL;
+	FuUsbEndpoint *ep2 = NULL;
+	g_autoptr(GPtrArray) ifaces = NULL;
+	g_autoptr(GPtrArray) endpoints = NULL;
+
+	ifaces = fu_usb_device_get_interfaces(FU_USB_DEVICE(self), error);
+	if (ifaces == NULL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_FOUND,
+				    "update interface not found");
+		return FALSE;
+	}
+	for (guint8 i = 0; i < ifaces->len; i++) {
+		iface = g_ptr_array_index(ifaces, i);
+		if (fu_usb_interface_get_class(iface) == FU_USB_CLASS_HID) {
+			endpoints = fu_usb_interface_get_endpoints(iface);
+			if (NULL == endpoints || endpoints->len == 0)
+				continue;
```

Comment:
`endpoints->len < 2` perhaps? Otherwise `g_ptr_array_index(endpoints, 1)` is going to explode if it only has one endpoint.